Cabal/tests/PackageTests/Tests.hs

```diff
@@ -120,10 +120,6 @@ tests config = do
   -- with the same name and LATER version
   tc "BuildDeps/InternalLibrary3" $ internal_lib_test "internal"
 
-  -- Test that an explicit dependency constraint which doesn't
```

Comment:
What exactly do you want here? There are several levels of strictness we could apply:

1. If ANY version bound is associated with an internal dep, error (because it doesn't do anything)
2. If the version bound is incompatible with the current package version, error (this would pick out people who were using the old feature)
3. Just ignore the version bound (so we still depend on the internal lib), but warn about it in cabal check

It seems better for this to be a cabal check test than a hard failure.